packages/contracts-bedrock/scripts/deploy/DeployAltDA.s.sol

```diff
@@ -0,0 +1,197 @@
+        return _challengeContractOwner;
+    }
+
+    function challengeWindow() public view returns (uint256) {
+        require(_challengeWindow != 0, "DeployAltDAInput: challengeWindow not set");
+        return _challengeWindow;
+    }
+
+    function resolveWindow() public view returns (uint256) {
+        require(_resolveWindow != 0, "DeployAltDAInput: resolveWindow not set");
+        return _resolveWindow;
+    }
+
+    function bondSize() public view returns (uint256) {
+        require(_bondSize != 0, "DeployAltDAInput: bondSize not set");
+        return _bondSize;
+    }
+
+    function resolverRefundPercentage() public view returns (uint256) {
+        require(_resolverRefundPercentage != 0, "DeployAltDAInput: resolverRefundPercentage not set");
+        return _resolverRefundPercentage;
+    }
+}
+
+contract DeployAltDAOutput is BaseDeployIO {
+    IDataAvailabilityChallenge internal _dataAvailabilityChallengeProxy;
+    IDataAvailabilityChallenge internal _dataAvailabilityChallengeImpl;
+
+    function set(bytes4 _sel, address _addr) public {
+        require(_addr != address(0), "DeployAltDAOutput: cannot set zero address");
+        if (_sel == this.dataAvailabilityChallengeProxy.selector) {
+            _dataAvailabilityChallengeProxy = IDataAvailabilityChallenge(payable(_addr));
+        } else if (_sel == this.dataAvailabilityChallengeImpl.selector) {
+            _dataAvailabilityChallengeImpl = IDataAvailabilityChallenge(payable(_addr));
+        } else {
+            revert("DeployAltDAOutput: unknown selector");
+        }
+    }
+
+    function dataAvailabilityChallengeProxy() public view returns (IDataAvailabilityChallenge) {
+        DeployUtils.assertValidContractAddress(address(_dataAvailabilityChallengeProxy));
+        return _dataAvailabilityChallengeProxy;
+    }
+
+    function dataAvailabilityChallengeImpl() public view returns (IDataAvailabilityChallenge) {
+        DeployUtils.assertValidContractAddress(address(_dataAvailabilityChallengeImpl));
+        return _dataAvailabilityChallengeImpl;
+    }
+}
+
+contract DeployAltDA is Script {
+    function run(DeployAltDAInput _dai, DeployAltDAOutput _dao) public {
+        deployDataAvailabilityChallengeProxy(_dai, _dao);
+        deployDataAvailabilityChallengeImpl(_dai, _dao);
+        initializeDataAvailabilityChallengeProxy(_dai, _dao);
+
+        checkOutput(_dai, _dao);
+    }
+
+    function deployDataAvailabilityChallengeProxy(DeployAltDAInput _dai, DeployAltDAOutput _dao) public {
+        bytes32 salt = _dai.salt();
+        vm.broadcast(msg.sender);
+        IProxy proxy = IProxy(
+            DeployUtils.create2({
+                _name: "Proxy",
+                _salt: salt,
+                _args: DeployUtils.encodeConstructor(abi.encodeCall(IProxy.__constructor__, (msg.sender)))
+            })
+        );
+        _dao.set(_dao.dataAvailabilityChallengeProxy.selector, address(proxy));
```

Comment:
I've been using [`vm.label`](https://book.getfoundry.sh/cheatcodes/label) from foundry as a way to make the test output nicer. See [here](https://github.com/ethereum-optimism/optimism/blob/6fbf9db6d57158797127516bc231dad480918289/packages/contracts-bedrock/scripts/deploy/DeploySuperchain.s.sol#L383).